crates/subspace-node/src/genesis_allocations.rs

```diff
@@ -0,0 +1,35 @@
+use subspace_runtime_primitives::{AccountId, Balance};
+use sp_core::crypto::Ss58Codec;
+use serde_json::Value;
+use sp_runtime::traits::Zero;
+
+pub fn get_genesis_allocations() -> Vec<(AccountId, Balance)> {
+    let file_content = include_str!("genesis_allocations.json");
+    
+    let allocations: Value = serde_json::from_str(file_content)
+        .expect("Failed to parse genesis allocations JSON");
+
+    allocations.as_array()
```

Comment:
All of these types implement Serde. You can just define the shape and then decode the whole thing without manually iterating over it, for balances you can use `NonZeroU128` to automatically check the invariant.

That is unless you want to have nice error messages here (that should never really happen anyway).